Meme/Meme/Models/TemplateExample.swift

```diff
@@ -0,0 +1,14 @@
+//
+//  TemplateExample.swift
+//  Meme
+//
+//  Created by Mesfin Bekele Mekonnen on 7/29/19.
+//  Copyright © 2019 Mesfin Bekele Mekonnen. All rights reserved.
+//
+
+import Foundation
+
+struct TemplateExample {
+    let name: String
+    let exampleURL: URL
```

Comment:
Because of how the Meme API is designed, for a given template in order to get the imageURL, you have to click through an example URL, which returns another dictionary of the following format : 
```{
    "direct": {
        "visible": "https://memegen.link/tenguy/your_text/goes_here.jpg",
        "masked": "https://memegen.link/_dGVuZ3V5CXlvdXJfdGV4dC9nb2VzX2hlcmUJ.jpg"
    },
    "markdown": {
        "visible": "![tenguy](https://memegen.link/tenguy/your_text/goes_here.jpg)",
        "masked": "![tenguy](https://memegen.link/_dGVuZ3V5CXlvdXJfdGV4dC9nb2VzX2hlcmUJ.jpg)"
    }
}```

Reply:
I then access the visible key in the direct dictionary, to get the image URL. :(. Very roundabout and a not nice hierarchy, But a template example is a layer that lives between a template and a Meme.

Reply:
you can probably get away with using the following:
```swift
urlString = "https://memegen.link/\(alias)/_.jpg"
```

means youll have to get the set of aliases from the previous request

Reply:
So, another way is to on _only_ kick of the templates api request which returns that dictionary of `{ name: templateURLString }`

but instead of going through the templateURLString, and knowing that the image path is `https://memegen.link/{some-identifier}/_.jpg`, you can build that url using the returned `templateURLString` mentioned above.
```swift
private let memegenDomain = "https://memegen.link/"

func imageURL(with id: String) -> URL? {
    return URL(string: memegenDomain + "\(id)/_.jpg")
}
...some other func
guard let templateIdentifier = URL(string: templateURLString)?.pathComponents.last
    let imageURL = imageURL(with: templateIdentifier) 
    { return nil }
...

```

Reply:
this would mean you just need one request

Reply:
V, you're a genius!